cmake add_subdirectory error.

Hi,

I'm building gl3w adding the following line to my CMakeLists.txt

add_subdirectory(gl3w)

but I've got the following error:

CMake Error: File /home/user/testdit/gl3w/cmake/Project-config.cmake does not exist. 
CMake Error at gl3w/CMakeLists.txt:65 (configure_file):
  configure_file Problem configuring file

I can reproduce the issue creating a directory tree like the following

testdir/gl3w
testdir/CMakeLists.txt
testdir/build

CMakeLists.txt has only one line
add_subdirectory(gl3w)

from the build directory
invoking
cmake ..

will generate the error

Any idea on why this is happening?

I'm using cmake 3.7.1

Thanks a lot!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,35 +43,35 @@ add_custom_target(
 )
 
 # create gl3w target
-add_library(${CMAKE_PROJECT_NAME} INTERFACE)
+add_library(${PROJECT_NAME} INTERFACE)
 
 # make gl3w target depend on the generator target
-add_dependencies(${CMAKE_PROJECT_NAME} gl3w_gen)
+add_dependencies(${PROJECT_NAME} gl3w_gen)
 
 # let remote project know about source and header files
-target_sources(${CMAKE_PROJECT_NAME} INTERFACE ${SOURCE_FILES})
-target_include_directories(${CMAKE_PROJECT_NAME} INTERFACE
+target_sources(${PROJECT_NAME} INTERFACE ${SOURCE_FILES})
+target_include_directories(${PROJECT_NAME} INTERFACE
 	$<BUILD_INTERFACE:${HEADER_DIR}>
 	$<INSTALL_INTERFACE:include>
 )
-target_include_directories(${CMAKE_PROJECT_NAME} INTERFACE ${EXTERNAL_INCLUDE_DIRS})
+target_include_directories(${PROJECT_NAME} INTERFACE ${EXTERNAL_INCLUDE_DIRS})
 # let remote project know which libraries need to be linked
-target_link_libraries(${CMAKE_PROJECT_NAME} INTERFACE ${EXTERNAL_LIBRARIES} dl)
+target_link_libraries(${PROJECT_NAME} INTERFACE ${EXTERNAL_LIBRARIES} dl)
 
 set(MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 set(BUILD_CMAKE_DIR "${CMAKE_CURRENT_BINARY_DIR}/cmake")
 
 # export targets for remote projects (i.e. make find_package(gl3w) work)
 configure_file(
-	"${MODULE_PATH}/${CMAKE_PROJECT_NAME}-config.cmake"
-	"${BUILD_CMAKE_DIR}/${CMAKE_PROJECT_NAME}-config.cmake"
+	"${MODULE_PATH}/${PROJECT_NAME}-config.cmake"
+	"${BUILD_CMAKE_DIR}/${PROJECT_NAME}-config.cmake"
 	COPYONLY
 )
 
 export(
-	TARGETS ${CMAKE_PROJECT_NAME}
-	FILE "${BUILD_CMAKE_DIR}/${CMAKE_PROJECT_NAME}-targets.cmake"
+	TARGETS ${PROJECT_NAME}
+	FILE "${BUILD_CMAKE_DIR}/${PROJECT_NAME}-targets.cmake"
 )
 
-export(PACKAGE ${CMAKE_PROJECT_NAME})
+export(PACKAGE ${PROJECT_NAME})
 

